src/ui/public/timepicker/kbn_global_timepicker.js

```diff
@@ -22,14 +22,20 @@ import { once, clone } from 'lodash';
 
 import toggleHtml from './kbn_global_timepicker.html';
 import { timeNavigation } from './time_navigation';
+import { timefilter } from 'ui/timefilter';
 
 uiModules
   .get('kibana')
-  .directive('kbnGlobalTimepicker', (timefilter, globalState, $rootScope) => {
-    const listenForUpdates = once($scope => {
-      $scope.$listen(timefilter, 'update', () => {
-        globalState.time = clone(timefilter.time);
-        globalState.refreshInterval = clone(timefilter.refreshInterval);
+  .directive('kbnGlobalTimepicker', (globalState, $rootScope) => {
+    const listenForUpdates = once(($scope) => {
+      $scope.$listen(timefilter, 'refreshIntervalUpdate', () => {
```

Comment:
Warning: `timefilter.emit('refreshIntervalUpdate')` may or may not happen within a digest cycle, meaning these and all other `$scope.$listen` calls need to ensure their handlers happen within a digest cycle... Maybe it's worth adding that logic to [`$listen`](https://github.com/elastic/kibana/blob/master/src/ui/public/listen/listen.js#L34)?